### PR TITLE
feat(eslint-plugin): no-unnecessary-condition

### LINKED ISSUE #115: [RFC]: @typescript-eslint:no-unnecessary-condition
### Why?

This rule avoid unnecessary and misleading code. See rule docs.

### Rule Documentation 📜

https://typescript-eslint.io/rules/no-unnecessary-condition/

### Config Selection 🛠

`recommended`

### (Optional) Additional Details 📝

_No response_

### PR DESCRIPTION
**Pull request by AntoineThibi**
Closes #115 

- Add the no-unnecessary-condition rule
- Improve the readme for starting up the project

### CODE REVIEW

File: packages/eslint-plugin/docs/Testing.md [hunk 1]
```diff
@@ -10,6 +10,14 @@ The example-app folder is located at the root of `react-native-project-config`.
 The goal of each file of example-app is to simulate the behavior of a rule or a set of rules, so that you can visually check it or by running the `yarn lint` command.
 When you add or remove a rule from the recommended config, please modify (or add/remove) the corresponding file in the example-app folder.
 
+## Adding a new rule
+
+In order to add a new rule, you must add it to the packages eslint-plugin, and then build it (it is in typescript) before seeing the changes in the example-app with the command :
```
Comment (jcalixte) [inline packages/eslint-plugin/docs/Testing.md:15]:
```suggestion
In order to add a new rule, you must add it to the packages eslint-plugin, and then build it (it is in typescript) before seeing the changes in the example-app with the command:
```

File: README.md [hunk 2]
```diff
@@ -19,6 +19,10 @@ There is also an example app under `packages/example-app` whose goal is to demon
 
 - clone the project: `git clone https://github.com/bamlab/react-native-project-config`,
 - install the dependencies: `yarn install`,
+- install the dependencies in the example-app: `cd example-app && yarn && cd ../`
```
Comment (alexisdelage) [inline README.md:22]: The example-app is already included in the workspace, so its dependancies are already installed with the first `yarn install`

File: README.md [hunk 3]
```diff
@@ -19,6 +19,10 @@ There is also an example app under `packages/example-app` whose goal is to demon
...
+- build the eslint-plugin: `cd packages/eslint-plugin && yarn build && cd ../..`
```
Comment (alexisdelage) [inline README.md:23]: You can replace with one command: `yarn workspace @bam.tech/eslint-plugin build`

File: example-app/eslint-breaking-examples/break-typescript-no-unnecessary-condition.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// Save without formatting: [⌘ + K] > [S]
+
+// This should trigger one error breaking eslint-plugin-react-native:
+// react-native-a11y/has-valid-accessibility-descriptors
```
Comment (alexisdelage) [inline example-app/eslint-breaking-examples/break-typescript-no-unnecessary-condition.ts:4]:
```suggestion
// @typescript-eslint/no-unnecessary-condition
```

File: packages/eslint-plugin/docs/Testing.md [hunk 5]
```diff
@@ -10,6 +10,14 @@ The example-app folder is located at the root of `react-native-project-config`.
...
+
+```bash
+cd packages/eslint-plugin && yarn build && cd ../..
```
Comment (alexisdelage) [inline packages/eslint-plugin/docs/Testing.md:18]: You can also use workspace command here

**Review by alexisdelage — APPROVED** [review]

**Review by matthieugicquel — APPROVED** [review]

**Pull request merged**